crates/typst-library/src/meta/bibliography.rs

```diff
@@ -445,6 +481,144 @@ impl Works {
     }
 }
 
+/// Converts a hayagriva title into a typst dictionary.
+fn title_to_dict(title: &hayagriva::types::Title) -> Dict {
+    dict! {
+        "canonical" => &*title.canonical.value,
+        "shorthand" => title.shorthand.as_ref().map(|f| &*f.value),
+        "translated" => title.translated.as_ref().map(|f| &*f.value),
+    }
+}
+
+/// Converts a hayagriva person into a typst dictionary.
+fn person_to_dict(person: &hayagriva::types::Person) -> Dict {
+    dict! {
+        "name" => &*person.name,
+        "given-name" => person.given_name.as_deref(),
+        "prefix" => person.prefix.as_deref(),
+        "suffix" => person.suffix.as_deref(),
+        "alias" => person.alias.as_deref(),
+    }
+}
+
+/// Converts an array of hayagriva persons into a typst array.
+fn persons_to_array(persons: &[hayagriva::types::Person]) -> Array {
+    persons.iter().map(|p| Value::Dict(person_to_dict(p))).collect()
+}
+
+/// Converts a hayagriva affiliated persons structure into a typst array.
+///
+/// Note that this makes a minor modification to the hayagriva structure
+/// to make ergonomics slightly better for a dynamically typed language -
+/// the persons and role are presented as a dictionary instead of a tuple.
+fn affiliated_persons_to_array(
+    affiliated_persons: &[(
+        Vec<hayagriva::types::Person>,
+        hayagriva::types::PersonRole,
+    )],
+) -> Array {
+    affiliated_persons
+        .iter()
+        .map(|(persons, role)| {
+            Value::Dict(dict! {
+                "persons" => persons_to_array(persons),
+                "role" => role.to_string(),
+            })
+        })
+        .collect()
+}
+
+/// Converts a hayagriva date (or something which can borrow as one) into a typst dictionary.
+fn date_to_dict(date: impl std::borrow::Borrow<hayagriva::types::Date>) -> Dict {
+    let date = date.borrow();
+    dict! {
+        "year" => date.year,
+        "month" => date.month,
+        "day" => date.day,
+    }
+}
```

Comment:
Shouldn't we use typst's `datetime` type here?

Reply:
Definitely, I did this before that was in typst IIRC.

Reply:
Yeah, makes sense haha. This is looking pretty good either way. Take your time!